bson/bsoncodec/default_value_decoders.go

```diff
@@ -452,17 +627,22 @@ func (dvd DefaultValueDecoders) Decimal128DecodeValue(dctx DecodeContext, vr bso
 		return fmt.Errorf("cannot decode %v into a decimal.Decimal128", vr.Type())
 	}
 
-	target, ok := i.(*decimal.Decimal128)
-	if !ok || target == nil {
+	switch target := i.(type) {
+	case *decimal.Decimal128:
+		d128, err := vr.ReadDecimal128()
```

Comment:
Is there a reason you moved the `ReadDecimal128` method invocation here? It adds repetitive code and extra errors to check.

Reply:
changed EmptyInterfaceDecodeValue's closure style